Add Qodana License Audit Documentation (#38)
JetBrains/Qodana: topics/qodana-yaml.md

## License Audit Configuration

Use license identifiers from [SPDX License List](https://spdx.org/licenses/).

### Exclude inspections
By default, Qodana License Audit includes all supported inspections.

Exclude inspections from analysis:

```yaml
version: 1.0
profile:
  name: qodana.recommended
exclude:
  - name: NoDependencyLicenses
  - name: ProhibitedDependencyLicense
  - name: UncategorizedDependencyLicense
  - name: UnrecognizedDependencyLicense
  - name: UnrecognizedProjectLicense
```

### Allowed and prohibited licenses

Override predefined list of licences:

```yaml
inspections:
  LicenseAudit:
    rules:
      - key:
          - "PROPRIETARY-LICENSE"
          - "MIT"
        prohibited:
          - "Apache-2.0"
        allowed:
          - "MIT"
          - "BSD-3-Clause"
          - "ISC"

      - key: "Apache-2.0"
        prohibited:
          - "MIT"
```

where `key` is project license, dependency license specified in `allowed` or `prohibited`.

### Override dependency license

> Override dependency license if you are not agree with detected dependency license:

Reviewer: ```suggestion
Override a dependency license if it has been detected incorrectly:
```

```yaml
inspections:
 LicenseAudit:
   dependencyOverrides:
       - name: "numpy"
         version: "1.19.1"
         licenses:
           - "BSD-3-Clause"
```

where `name` is dependency name, `version` is dependency version and `licenses` - list of new redefined dependency licenses used by License Audit.

### Ignore dependency license

> Ignore dependency license if you do not want to see problems connected with this dependency license:

Reviewer: ```suggestion
Ignore a dependency license to hide the related problems from the view:
```
